boards/components/src/appid/checker_signature.rs

```diff
@@ -10,50 +10,63 @@ use kernel::hil::{digest, public_key_crypto};
 
 #[macro_export]
 macro_rules! app_checker_signature_component_static {
-    ($S:ty, $H:ty, $HL:expr, $SL:expr $(,)?) => {{
-        let hash_buffer = kernel::static_buf!([u8; $HL]);
-        let signature_buffer = kernel::static_buf!([u8; $SL]);
+    ($SignatureKind:ty, $HashKind:ty, $HASH_LEN:expr, $SIGNATURE_LEN:expr $(,)?) => {{
+        let hash_buffer = kernel::static_buf!([u8; $HASH_LEN]);
+        let signature_buffer = kernel::static_buf!([u8; $SIGNATURE_LEN]);
         let checker = kernel::static_buf!(
             capsules_system::process_checker::signature::AppCheckerSignature<
                 'static,
-                $S,
-                $H,
-                $HL,
-                $SL,
+                $SignatureKind,
+                $HashKind,
+                $HASH_LEN,
+                $SIGNATURE_LEN,
             >
         );
 
         (checker, hash_buffer, signature_buffer)
     };};
 }
 
-pub type AppCheckerSignatureComponentType<S, H, const HL: usize, const SL: usize> =
-    capsules_system::process_checker::signature::AppCheckerSignature<'static, S, H, HL, SL>;
+pub type AppCheckerSignatureComponentType<
+    SignatureKind,
+    HashKind,
```

Comment:
Might be nitpicky, but these fields don't really refer to a `SignatureKind` "marker" as much as they refer to a `Signer` "component" that actually does work, right? Same for `HashKind` -> `Hasher`?

I'm slightly confused by the `Kind` part in the generic's name, but it's not a big deal either way.

Reply:
This is a good nitpick to discuss, the whole point is clarity from more explicit names.

This is also definitely a case of Pat might be learning Rust via PR discussions... but my rationale for the current naming was that, IIUC, this is ultimately the placeholder label for a type. E.g., looking at [the documentation explaining bounds](https://doc.rust-lang.org/rust-by-example/generics/bounds.html) this would be a "generic type `SignatureKind`". I used "Kind" in lieu of "Type" as I thought that was a common convention for avoiding complexity around "Type" as a keyword, though maybe it is not needed here.

Would `SignerType` (and `HasherType`) make more sense?